token/program-2022/src/extension/reallocate.rs

```diff
@@ -0,0 +1,87 @@
+use {
+    crate::{
+        error::TokenError,
+        extension::{AccountType, ExtensionType, StateWithExtensions, StateWithExtensionsMut},
+        processor::Processor,
+        state::Account,
+    },
+    solana_program::{
+        account_info::{next_account_info, AccountInfo},
+        entrypoint::ProgramResult,
+        msg,
+        program::invoke,
+        pubkey::Pubkey,
+        system_instruction,
+        sysvar::{rent::Rent, Sysvar},
+    },
+};
+
+/// Processes a [Reallocate](enum.TokenInstruction.html) instruction
+pub fn process_reallocate(
+    program_id: &Pubkey,
+    accounts: &[AccountInfo],
+    new_extension_types: Vec<ExtensionType>,
+) -> ProgramResult {
+    let account_info_iter = &mut accounts.iter();
+    let token_account_info = next_account_info(account_info_iter)?;
+    let payer_info = next_account_info(account_info_iter)?;
+    let system_program_info = next_account_info(account_info_iter)?;
+    let authority_info = next_account_info(account_info_iter)?;
+    let authority_info_data_len = authority_info.data_len();
+
+    // check that account is the right type and validate owner
+    let mut current_extension_types = {
+        let token_account = token_account_info.data.borrow();
+        let account = StateWithExtensions::<Account>::unpack(&token_account)?;
```

Comment:
This only supports Account reallocation atm. Do we need to support Mint reallocation for anything, eg ConfidentialTransfers?

Reply:
I don't have a mint reallocation use case in mind.  Keeping the mint configuration static after creation has a strong appeal to me right now, not eager to open a new rugging vector